tests/integration/dhcp_test.py

```diff
@@ -204,18 +560,74 @@ def _remove_veth_pair():
 def _setup_dhcp_nics():
     assert libcmd.exec_cmd(['ip', 'link', 'set', DHCP_SRV_NIC, 'up'])[0] == 0
     assert libcmd.exec_cmd(['ip', 'link', 'set', DHCP_CLI_NIC, 'up'])[0] == 0
-    assert libcmd.exec_cmd(['ip', 'addr', 'add', IPV4_ADDRESS1, 'dev',
-                            DHCP_SRV_NIC])[0] == 0
+    assert libcmd.exec_cmd(['ip', 'addr', 'add', "{}/24".format(DHCP_SRV_IP4),
+                            'dev', DHCP_SRV_NIC])[0] == 0
+    with open(SYSFS_DISABLE_RA_SRV, 'w') as fd:
+        fd.write('0')
+
+    # This stop dhcp server NIC get another IPv6 address from radvd.
     with open(SYSFS_DISABLE_IPV6_FILE, 'w') as fd:
         fd.write('0')
-    assert libcmd.exec_cmd(['ip', 'addr', 'add', IPV6_ADDRESS1, 'dev',
-                            DHCP_SRV_NIC])[0] == 0
+
+    assert libcmd.exec_cmd(['ip', 'addr', 'add', "{}/64".format(DHCP_SRV_IP6),
+                            'dev', DHCP_SRV_NIC])[0] == 0
 
 
 def _clean_up():
     libcmd.exec_cmd(['systemctl', 'stop', 'dnsmasq'])
+    libcmd.exec_cmd(['systemctl', 'stop', 'radvd'])
     _remove_veth_pair()
     try:
         os.unlink(DNSMASQ_CONF_PATH)
     except FileNotFoundError:
         pass
+    except OSError:
+        pass
+    try:
+        os.unlink(RADVD_CONF_PATH)
+    except FileNotFoundError:
+        pass
+    except OSError:
+        pass
+
+
+def _get_dns_conf():
+    """
+    Return a list of name server string configured in /etc/resolv.conf
+    """
+    ret = []
+    regex = re.compile('^nameserver +([0-9\.a-f:]+)$')
+    with open(RESOLV_CONF_PATH, 'r') as fd:
+        for line in fd:
+            match = regex.match(line)
+            if match:
+                ret.append(match.group(1))
+    return ret
+
+
+def _get_ipv4_main_routes():
+    """
+    return a list of route entry string from `ip -4 route`.
+    """
+    rc, output, _ = libcmd.exec_cmd(['ip', '-4', 'route'])
+    if rc != 0:
+        return []
+    return [line for line in output.split('\n') if line]
+
+
+def _get_ipv6_main_routes():
+    """
+    return a list of route entry string from `ip -6 route`.
+    """
+    rc, output, _ = libcmd.exec_cmd(['ip', '-6', 'route'])
+    if rc != 0:
+        return []
+    return [line for line in output.split('\n') if line]
+
+def _has_ipv6_dns():
```

Comment:
Maybe rename to `_has_ipv6_nameserver()`.

Reply:
Done. Renamed to `_has_ipv6_auto_nameserver`.